src/lib.rs

```diff
@@ -171,7 +174,7 @@ pub fn validate() {
 /// The encryption key will be written to the `keys.json` file at the root of your local secrets repository. You will need to commit this change yourself.
 #[no_mangle]
 pub fn generate_encryption_key() -> String {
-    crate::encryption::generate_key()
+    crate::encryption::generate_key().to_string()
```

Comment:
I am trying to understand some of the syntax choices you have made so when I hopefully contribute to this project, I can follow the same choices. I can see that in some cases such as [here](https://github.com/Automattic/configure/blob/add/single-file-encryption-and-decryption/src/configure.rs#L1) you have opted to use `*` namespace and then called the [`decrypt_files_for_configuration` function directly](https://github.com/Automattic/configure/blob/add/single-file-encryption-and-decryption/src/configure.rs#L230). In this snippet however, you have opted to go with the most verbose approach of `crate::{abc}`.

Is that an implementation specific approach where you felt that some of them needed to be more verbose? Or is it maybe an oversight? I can see that there is some consistency in between some calls, so maybe you just followed what you've started with, possibly bringing code from various documentation which used different approaches?

[From the book, it looks like either is valid depending on project & preference](https://doc.rust-lang.org/book/ch07-03-paths-for-referring-to-an-item-in-the-module-tree.html).

P.S: I personally always used relative path with medium verbosity, where I didn't use the `*` namespace and imported stuff directly so when I do `find next usage` in VIM, it'll either bring it to me the local implementation or the import. With `go to definition` it's not a huge deal either way, but I liked that middle ground.

Reply:
I'm probably a little all over the place on this, but my heuristic is generally:

1. For third-party dependencies, explicitly name every import.
2. For first-party dependencies (inside the same project), if we only need to import one thing, name it. If we need to import more than one thing, use `*`. 

I'm not certain I've stuck to this – I kinda wish there were just a linter setting we could make for this – I don't feel strongly about any particular approach (though manually specifying paths seems...less ideal IMHO).

Reply:
In the projects I worked on, I liked always using one namespace (such as `encryption::generate_key`) and as long as that namespace is not too generic, it was obvious whether it belonged to the project or to a 3rd party library. When it wasn't obvious, I'd opt to include one more for 3rd party libraries.

I haven't worked on this project at all, so it's hard to say whether that'd work well here, but when/if I do work on the project, I could maybe create a small PR for us to discuss.